infra/modules/azure_api_management/network.tf

```diff
@@ -0,0 +1,81 @@
+# Define the A Records for APIMv2
+
+resource "azurerm_private_dns_a_record" "apim_azure_api_net" {
+  name                = azurerm_api_management.this.name
+  zone_name           = data.azurerm_private_dns_zone.azure_api_net.name
+  resource_group_name = data.azurerm_virtual_network.this.resource_group_name
+  ttl                 = 3600
+  records             = azurerm_api_management.this.private_ip_addresses
+
+  tags = var.tags
+}
+
+resource "azurerm_private_dns_a_record" "apim_management_azure_api_net" {
+  name                = azurerm_api_management.this.name
+  zone_name           = data.azurerm_private_dns_zone.management_azure_api_net.name
+  resource_group_name = data.azurerm_virtual_network.this.resource_group_name
+  ttl                 = 3600
+  records             = azurerm_api_management.this.private_ip_addresses
+
+  tags = var.tags
+}
+
+resource "azurerm_private_dns_a_record" "apim_scm_azure_api_net" {
+  name                = azurerm_api_management.this.name
+  zone_name           = data.azurerm_private_dns_zone.scm_azure_api_net.name
+  resource_group_name = data.azurerm_virtual_network.this.resource_group_name
+  ttl                 = 3600
+  records             = azurerm_api_management.this.private_ip_addresses
+
+  tags = var.tags
+}
+
+# Link A Records into the VNet
+
+resource "azurerm_private_dns_zone_virtual_network_link" "azure_api_link" {
+  name                  = format("%s-az-link", local.apim.name)
+  resource_group_name   = data.azurerm_virtual_network.this.resource_group_name
+  private_dns_zone_name = data.azurerm_private_dns_zone.azure_api_net.name
+  virtual_network_id    = data.azurerm_virtual_network.this.id
+}
+
+
+resource "azurerm_private_dns_zone_virtual_network_link" "management_api_link" {
+  name                  = format("%s-mngt-link", local.apim.name)
+  resource_group_name   = data.azurerm_virtual_network.this.resource_group_name
+  private_dns_zone_name = data.azurerm_private_dns_zone.management_azure_api_net.name
+  virtual_network_id    = data.azurerm_virtual_network.this.id
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "scm_apim_link" {
+  name                  = format("%s-scm-link", local.apim.name)
+  resource_group_name   = data.azurerm_virtual_network.this.resource_group_name
+  private_dns_zone_name = data.azurerm_private_dns_zone.scm_azure_api_net.name
+  virtual_network_id    = data.azurerm_virtual_network.this.id
+}
```

Comment:
The link of these dns zones to the vnet should only be done once per zone/vnet. For this reason I think we shouldn't replicate them in this module

```suggestion
```